billing/checkout/service.go

```diff
@@ -456,6 +460,31 @@ func (s *Service) SyncWithProvider(ctx context.Context, customerID string) error
 		if checks[idx], err = s.repository.UpdateByID(ctx, ch); err != nil {
 			return fmt.Errorf("failed to update checkout session: %w", err)
 		}
+
+		if checkoutSession.CustomerDetails != nil && checkoutSession.CustomerDetails.Address != nil {
```

Comment:
Instead of syncing these details here, I think we should add a `cron` syncer just like we have for checkouts/subscriptions if we want to keep customer details in sync always. This is not done for customers at the moment.